tests/e2e/network/multiple_standard_lb.go

```diff
@@ -235,6 +238,34 @@ var _ = Describe("Ensure LoadBalancer", Label(utils.TestSuiteLabelMultiSLB), fun
 	})
 })
 
+func getDeploymentPodsNodeNames(kubeClient clientset.Interface, namespace, deploymentName string) (map[string]bool, error) {
+	var (
+		podList *v1.PodList
+		res     = make(map[string]bool)
+		err     error
+	)
+	err = wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 5*time.Minute, false, func(ctx context.Context) (bool, error) {
+		podList, err = kubeClient.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, pod := range podList.Items {
+			if strings.HasPrefix(pod.Name, deploymentName) && pod.Spec.NodeName == "" {
+				utils.Logf("Waiting for pod %s to be running", pod.Name)
+				return false, nil
+			}
+			utils.Logf("Pod %s is running on node %s", pod.Name, pod.Spec.NodeName)
+			res[pod.Spec.NodeName] = true
```

Comment:
If there're Pods not from this deployment, will them be put into res as well?

Reply:
There should not be other pods, but I didn't notice that we don't stop the process if failed to delete host exec pod, so there could be other pods.

Reply:
Since the function is called `getDeploymentPodsNodeNames`, I think it is better to only consider Pods of the deployment.

Reply:
yes